Main.hs

```diff
@@ -229,18 +246,28 @@ compileTerm :: Builtins -> Term -> TCM (Hs.Exp ())
 compileTerm builtins v =
   case unSpine v of
     Var x es   -> (`app` es) . Hs.Var () . Hs.UnQual () . hsName =<< showTCM (Var x [])
-    Def f es   -> (`app` es) . Hs.Var () =<< hsQName builtins f
+    Def f es  
+      | show (qnameName f) == "if_then_else_"
+      -> do [b, left, right] <- compileArgs es
```

Comment:
Here we should maybe fail more gracefully (or eta-expand!) on partially applied `if`, and not fail at all on over-application. For instance
```haskell
(if b then id else suc) 0
```